src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/WcfUserNameService.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IdentityModel.Selectors;
+using System.ServiceModel;
+
+namespace WcfService
+{
+    public class CustomUserNameValidator : UserNamePasswordValidator
+    {
+        // This method validates users. It allows in two users, test1 and test2 
+        // with passwords 1tset and 2tset respectively.
+        public override void Validate(string userName, string password)
+        {
+            if (null == userName || null == password)
+            {
+                throw new ArgumentNullException();
+            }
+
+            if (!(userName == "test1" && password == "1tset") && !(userName == "test2" && password == "2tset"))
```

Comment:
NIT: Not sure if `1tset` and `2tset` are typos or deliberate, but you might end up with some weird test bugs if someone misreads `tset` for `test` :smile: